### PR TITLE
add documentation for support custom claim mapping feature

### PR DESCRIPTION
**Pull request by tharmini**
## Purpose
Add documentation for support custom claim mapping feature

### CODE REVIEW

File: en/docs/how-tos/support-custom-claims-mapping.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Support Custom Claims Mapping
+
+When JWT retrieved from Multiple Identity providers, Microgateway can map the relevant claims as a supported one to validates the JWT.  The claims of the incoming authentication JWT can differ based on the Authorization server that issued the JWT token. This feature maps the incoming jwt remote claims to the local claims of the microgateway as well as the claims expected by the back end service.  You can use custom claims mapping transformation in micro gateway when you have different keys or values of claims in your JWT token to make it as a Microgateway supported one.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/support-custom-claims-mapping.md:3]: When JWT retrieved from Multiple Identity providers, Microgateway can map the relevant claims as a supported one to **validate** the JWT.  The claims of the incoming authentication JWT can differ based on the Authorization server that issued the JWT token. This feature maps the incoming **JWT** remote claims to the local claims of the **Microgateway** as well as the claims expected by the back end service.  You can use custom claims mapping transformation in micro gateway when you have different keys or values of claims in your JWT token to make it as a Microgateway supported one.

File: en/docs/how-tos/support-custom-claims-mapping.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     claimMapperClassName = "org.wso2.micro.gateway.jwtTransformer.DefaultJwtTransformer"
+     [[jwtTokenConfig.claims]]
+       remoteClaim = "scp"
+       localClaim = "scope" 
+```
+         
+      
+2. If the format of the remote claim value differs from the custom claims then you should write a class to transform it into the correct format and define the claim mapping class name in the config file.
+
+     The following subsections explain how you can work with claims value transformation. Here the existing scope value is in an array and it shoud be change to string with space.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/support-custom-claims-mapping.md:26]: The following subsections explain how you can work with claims value transformation. Here the existing scope value is in an array and it **should be changed** to string with space.

File: en/docs/how-tos/support-custom-claims-mapping.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+#### Writing JWT transformer for claims value
+Microgateway defines java interfaces to write the JWT value transformer. The interface for the transformer defined below. Developers can implement these interfaces to achieve custom transformation logic. 
+```` java
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/support-custom-claims-mapping.md:29]: Shall we add a git location of a sample project for this?

File: en/docs/how-tos/support-custom-claims-mapping.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        **JWT transformer dependency**
+        ``` java
+        <dependency>
+            <groupId>org.wso2.am.microgw</groupId>
+            <artifactId>mgw-jwt-transformer</artifactId>
+            <version>3.2.0</version>
+        </dependency>
+        ```
+#### Adding JWT transformer to the project 
+Once JWT claims value transformer is written, then the JWT  transformer project should be built and the output jar should be placed in the <MGW-project>/lib directory. If third-party libraries are used when writing jwt claims mapping transformer, these custom jars should also be placed in the same directory.
```
Comment (AmaliMatharaarachchi) [inline en/docs/how-tos/support-custom-claims-mapping.md:89]: Once JWT claims value transformer is written, then the JWT  transformer project should be built and the output jar should be placed in the <MGW-project>/lib directory. If third-party libraries are used when writing **JWT** claims mapping transformer, these custom jars should also be placed in the same directory.

**Comment by Mariangela** [comment]
@tharmini , please address the review comments.

File: en/mkdocs.yml [hunk 5]
```diff
@@ -108,6 +108,7 @@ nav:
         - Configure Caching: how-tos/configure-caching.md
         - Manage Microgateway APIs with WSO2 API Manager: how-tos/manage-microgateway-apis-with-wso2-api-manager.md
         - Adding Extension Points: how-tos/adding-extension-points.md
+        - support-custom-claims-mapping: how-tos/support-custom-claims-mapping.md
```
Comment (praminda) [inline en/mkdocs.yml:111]:
```suggestion
        - Support Custom Claims Mapping: how-tos/support-custom-claims-mapping.md
```

File: en/docs/how-tos/support-custom-claims-mapping.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+In a detailed view for an example if you generate JWT with claims with a different key like claim key is ‘scp’ and value is ‘[“write”, “read”]’. But in Microgateway it only validates correctly if the key is scope and values of scope separated by space as a string. Hence through the custom claims mapping feature, you can map the remote claims to local claims to support the Microgateway.  Here remote claims come from an external provider thus local claims need to be mapped.
+ 
+1. If the key of the remote claims is different from the custom claims, then you need to change the configuration as below.  You can define the multiple keys of the remote claim and local claim.
+
+     Adding the following section to the micro-gw.conf file which is located in the <MGW-RUNTIME-HOME>/conf directory, you can change the configurations in the jwtTokenConfig. Here the remote claim is scp and the local claim is scope.
```
Comment (praminda) [inline en/docs/how-tos/support-custom-claims-mapping.md:9]:
```suggestion
     Adding the following section to the `micro-gw.conf` file which is located in the `<MGW-RUNTIME-HOME>/conf` directory, you can change the configurations in the jwtTokenConfig. Here the remote claim is scp and the local claim is scope.
```

**Review by Rajith90 — APPROVED** [review]

**Comment by Rajith90** [comment]
We will send a separate PR addressing the changes required

**Pull request merged**